scattering/scattering.py

```diff
@@ -77,27 +84,64 @@ def structure_factor(trj, Q_range=(0.5, 50), n_points=1000, framewise_rdf=False)
 
             x_a = compositions[e1]
             x_b = compositions[e2]
-
-            pre_factor = x_a * x_b * f_a * f_b * 4 * np.pi * rho
-            try:
-                g_r = rdfs['{0}{1}'.format(e1, e2)]
-            except KeyError:
-                pairs = top.select_pairs(selection1='element {}'.format(e1),
-                                         selection2='element {}'.format(e2))
-                if framewise_rdf:
-                    r, g_r = rdf_by_frame(trj,
-                                         pairs=pairs,
-                                         r_range=(0, L / 2),
-                                         bin_width=0.001)
+            
+            if pair == None:
+                try:
+                    g_r = rdfs['{0}{1}'.format(e1, e2)]
+                except KeyError:
+                    pairs = top.select_pairs(selection1=top.select('element {}'.format(e1)),
+                                             selection2=top.select('element {}'.format(e2)))
+                    if framewise_rdf:
+                        r, g_r = rdf_by_frame(trj,
+                                             pairs=pairs,
+                                             r_range=(0, L / 2),
+                                             bin_width=0.001)
+                    else:
+                        r, g_r = md.compute_rdf(trj,
+                                                pairs=pairs,
+                                                r_range=(0, L / 2),
+                                                bin_width=0.001)
+                    rdfs['{0}{1}'.format(e1, e2)] = g_r
+                integral = simps(r ** 2 * (g_r - 1) * np.sin(q * r) / (q * r), r)
+                pre_factor = x_a * x_b * f_a * f_b * 4 * np.pi * rho
+            else:
+                n_element_1 = len(top_1.select('element {}'.format(e1)))
+                n_element_2 = len(top_2.select('element {}'.format(e2)))
+
+                if (length_1 == 0) or (length_2 == 0):
+                    integral = 0
+                    pre_factor = 0
                 else:
-                    r, g_r = md.compute_rdf(trj,
-                                            pairs=pairs,
-                                            r_range=(0, L / 2),
-                                            bin_width=0.001)
-                rdfs['{0}{1}'.format(e1, e2)] = g_r
-            integral = simps(r ** 2 * (g_r - 1) * np.sin(q * r) / (q * r), r)
-            num += pre_factor * integral + int(e1 == e2)
-        S[i] = num/denom
+                    try:
+                        g_r = rdfs['{0}{1}'.format(e1, e2)]
```

Comment:
I'm actually not sure what the preferred way is in term so adding key-value pairs to a dictionary.  The other way is using conditionals like this:
```
if ['{}-{}'.format(e1, e1)] in Number_scale:
    continue
else:
    pass
```

@mattwthompson do you have a preference?